.github/workflows/build.yml

```diff
@@ -64,75 +64,149 @@ jobs:
-            content-type: "application/zip"
-            public: true
+          aws-access-key-id: ${{ env.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ env.AWS_SECRET_ACCESS_KEY }}
+          aws-region: ${{ env.AWS_REGION }}
+          aws-bucket: ${{ env.BUCKET_NAME }}
+          bucket-root: "/"
+          destination-dir: "/"
+          file-path: "${{inputs.package-dist-directory}}/${{ env.ZIP_NAME }}"
+          content-type: "application/zip"
+          public: true
 
       # Create branch on Amplify
       - name: Create Amplify branch
-        if: github.event_name == 'pull_request' && inputs.deploy == true
+        if: inputs.deploy == true
         shell: bash
         # We return true to prevent the step from failing if the branch already exists
         run: |
-            aws amplify create-branch \
-            --app-id ${{ inputs.amplify-app-id }} \
+          aws amplify create-branch \
+            --app-id ${{ env.AMPLIFY_ID }} \
             --branch-name ${{ env.BRANCH_NAME }} \
             || true
 
       # Deploy Amplify from S3
       - name: Deploy Amplify from S3
-        if: github.event_name == 'pull_request' && inputs.deploy == true
+        if: inputs.deploy
         shell: bash
         run: |
-            aws amplify start-deployment \
-            --app-id ${{ inputs.amplify-app-id }} \
+          aws amplify start-deployment \
+            --app-id ${{ env.AMPLIFY_ID }} \
             --branch-name ${{ env.BRANCH_NAME }} \
             --source-url https://${{ env.BUCKET_NAME }}.s3.${{ env.AWS_REGION }}.amazonaws.com/${{ env.ZIP_NAME }}
+      # Set env vars for domain name association
+      - name: Set environment variables for domain name association
+        if: github.event_name == 'pull_request' && inputs.deploy
+        run: |
+          PACKAGE_NAME_SUFFIX=$(echo "${{ inputs.package-name }}" | sed -e 's/^-app//')
+          echo "SUB_DOMAIN=pr${{ github.event.number }}-aperture-$PACKAGE_NAME_SUFFIX" >> $GITHUB_ENV
+
+      - name: Get existing domain associations
+        if: github.event_name == 'pull_request' && inputs.deploy
+        run: |
+          DOMAIN_ASSOCIATIONS=$(aws amplify get-domain-association \
+            --app-id ${{ env.AMPLIFY_ID }} \
+            --domain-name pie.design \
+            --region ${{ env.AWS_REGION }} \
+            | jq '
+              [.domainAssociation.subDomains[].subDomainSetting | { prefix: (.prefix // ""), branchName }] |
+              . |= (map(.branchName) | index("pr${{ github.event.number }}")) as $ix |
+              if $ix == null then
+                . + [{branchName: "pr${{ github.event.number }}", prefix: "${{ env.SUB_DOMAIN }}"}]
+              else . end
+            ' -c)
+          echo "DOMAIN_ASSOCIATIONS=$DOMAIN_ASSOCIATIONS" >> $GITHUB_ENV
+      - name: Create Subdomain Association
+        if: github.event_name == 'pull_request' && inputs.deploy
```

Comment:
Do we still need the pull request check? Yes is a valid answer, just checking though 😄

Reply:
Yep, only because we don't need to do sub-domain logic for deploys to main 👍🏼